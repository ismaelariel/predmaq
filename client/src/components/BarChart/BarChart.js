import React from "react";

import { Chart as ChartJS } from "chart.js/auto";
import { ArcElement, Tooltip, Legend } from "chart.js";
import { Bar } from "react-chartjs-2";

ChartJS.register(ArcElement, Tooltip, Legend);

const BarChart = ({ predict, nameState }) => {
    const labelOne = {
        labels: predict.map((data) => data.ProductID),
        datasets: [
            {
                label: nameState,
                data: predict.map((data) => nameState === "Desgaste" ?
                    data.ToolWear : data.RotationalSpeed),
                backgroundColor: '#0000ff'
            }
        ]
    };

    const labelMany = {
        labels: predict.map((data) => data.ProductID),
        datasets: [
            {
                label: "Temperature",
                data: predict.map((data) => data.ProcessTemperature),
                backgroundColor: '#0000ff'
            },
            {
                label: "Desgaste",
                data: predict.map((data) => nameState === "Desgaste" ||
                    nameState === "Pesquisa por Modelo" ?
                    data.ToolWear :
                    data.RotationalSpeed),
                backgroundColor: '#ff0000'
            },
            {
                label: "Torque",
                data: predict.map((data) => data.Torque),
                backgroundColor: '#008000'
            }
        ]
    };

    return (
        <Bar
            data={predict.length === 1 ? labelMany : labelOne}
            width={100}
            height={100}
            options={{ maintainAspectRatio: false }}
        />
    );
};

export default BarChart;
