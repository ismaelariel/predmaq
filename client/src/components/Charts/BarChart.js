import React from "react";

import { Chart as ChartJS } from "chart.js/auto";
import { ArcElement, Tooltip, Legend } from "chart.js";
import { Bar } from "react-chartjs-2";

ChartJS.register(ArcElement, Tooltip, Legend);

const BarChart = ({ predict, nameState }) => {
    const labelOne = {
        labels: predict.map((data) => data.ProductID),
        datasets: [
            {
                label: nameState,
                data: predict.map((data) => nameState === "Desgaste" ?
                    data.ToolWear : data.RotationalSpeed)
            }
        ]
    };

    const labelMany = {
        labels: predict.map((data) => data.ProductID),
        datasets: [
            {
                label: "Temperature",
                data: predict.map((data) => data.ProcessTemperature),
                backgroundColor: '#96c2f2'
            },
            {
                label: "Desgaste",
                data: predict.map((data) => nameState === "Desgaste" ||
                    nameState === "Pesquisa por Modelo" ?
                    data.ToolWear :
                    data.RotationalSpeed),
                backgroundColor: '#f2a7c0'
            },
            {
                label: "Torque",
                data: predict.map((data) => data.Torque),
                backgroundColor: '#f1c2a4'
            }
        ]
    };

    return (
        <Bar
            data={predict.length === 1 ? labelMany : labelOne}
            width={100}
            height={100}
            options={{ maintainAspectRatio: false }}
        />
    );
};

export default BarChart;
