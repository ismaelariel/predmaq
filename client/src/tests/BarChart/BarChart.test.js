import React from "react";
import renderer from "react-test-renderer";
import machine from "../../utils/machine.json";

import BarChart from "../../components/BarChart/BarChart";

it("render linechart view", () => {
    const component = renderer.create(
        <BarChart
            predict={machine.data}
            nameState="Desgaste"
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
