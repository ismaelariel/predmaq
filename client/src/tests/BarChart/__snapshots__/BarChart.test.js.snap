// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render linechart view 1`] = `
<canvas
  height={100}
  role="img"
  width={100}
/>
`;
